### ISSUE TITLE
Affine Registration Support

### ISSUE THREAD

**Issue by J-Yash** [dipy/cudipy#2]
Hi,
I am using DIPY for some image registration. My current registration pipeline involves center of mass , translation, rigid, affine, and SymmetricDiffeomorphic registrations. From the CuDIPY readme, I understand that currently I can only implement the SymmetricDiffeomorphic registration in cudipy. Is that correct? Are you planning on adding gpu support to other registrations (affine) soon?

**Comment by grlee77** [comment]
Yes, we do hope to add it at some point, but I haven't had a chance to work on this repository recently. It is likely to take multiple days of effort to complete an implementation and I don't have a specific timeline for this at the moment. We would like to get funding to support dedicating time to this as I think it would be pretty useful to the community.

The main parts needed for this to happen are porting the Cython code in `dipy/align/parzenhist.pyx` to CUDA kernels and porting the L-BFGS-B optimizer from SciPy to a version that supports GPU arrays. The remaining code in `imaffine.py` itself should be easy to port to the GPU just by swapping NumPy->CuPy in appropriate places once those components are there.

**Comment by J-Yash** [comment]
Thank you for the response. While I do understand the challenges, it is great to hear that it is still on your to-do list. We are planning on using DIPY for our registration pipeline for our work. Since our pipeline involves running multiple registrations on large 3D images, GPU support would certainly be very welcome. I do hope to see progress soon. 

Thank you very much.

**Comment by ekurtulus** [comment]
Another future user here. It would be really nice if affine registration could be prioritized.

**Comment by J-Yash** [comment]
Hi, just wanted to check if there have been any updates on this issue?
